### PR TITLE
Remove unused problem variables

### LINKED ISSUE #27193: Remove unused variables of Problems interface
Open question:
- Does ValidateStep use the Worker API? If so, let's create a separate issue to migrate away from the JSON-based serialisation.

### PR DESCRIPTION
**Pull request by reinsch82**
Fixes 
 - #27193  

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by reinsch82** [comment]
@bot-gradle test this

**Review by donat — COMMENTED** [review]
LGTM

**Review by donat — APPROVED** [review]

File: platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/transform/IsolateTransformParametersCodec.kt [hunk 1]
```diff
@@ -60,8 +59,7 @@ class IsolateTransformParametersCodec(
             isolatableFactory,
             buildOperationExecutor,
             classLoaderHierarchyHasher,
-            fileCollectionFactory,
-            ownerService()
```
Comment (hegyibalint) [inline platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/transform/IsolateTransformParametersCodec.kt:64]: ❔Just interested, was this also found? I don't see the connection with the PAPI.

**Review by hegyibalint — APPROVED** [review]
I somehow see a lot of `compileAll` rounds to do all of this 😅 
Nice work; I am happy that this is cleaned up!

File: platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/transform/IsolateTransformParametersCodec.kt (hunk 1 above)
Comment (reinsch82) [reply platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/transform/IsolateTransformParametersCodec.kt:64]: The `IsolateTransformParameters` constructor took a `Problems` instance.
That got removed by IDEA refactoring when I removed the parameter.

**Comment by reinsch82** [comment]
@bot-gradle merge

**Pull request merged**